vm/src/obj/objbyteinner.rs

```diff
@@ -25,6 +25,18 @@ pub struct PyByteInner {
     pub elements: Vec<u8>,
 }
 
+pub fn vec_from_str(value: &str, encoding: &str, vm: &VirtualMachine) -> PyResult<Vec<u8>> {
```

Comment:
maybe implementing `from_string` to pybyteinner ?

Reply:
That was one of my early idea but then I found it is hard to turns `PyByteInner` into `PyBytes`. Do you have a good solution for that? For now, I think creating a vector is good enough for this kind of minimal charset transition API. Because it doesn't use any feature of `PyBytesInner` and easy to transit to it once it is required. Tell me if I missed something. I feel like I am very new to the str/bytes (actually for everything in this project, but especially).

Reply:
there is a try_from_object in #847 which creates a PyByteInner from a PyObjectRef(which could be bytes, byterray, memoryview). I think it make the API more consitent to use use a PyByteInner::from_string and the use PyByteInner.elements 
It's a personal opinion and I'm not in charge of this project.

Reply:
added PyBytes::from_string and PyByteInner::from_string